### PR TITLE
TagGroup beta API updates

### PR DESCRIPTION
**Pull request by tiffany-sy-lin**
- remove isDisabled and disabledKeys
- In onRemove, pass tag id instead of content
- remove event arg from onRemove
- Rename isRemovable to follow props guidelines: 'allowsX'
- Remove 'with announcing' story
- Confirm focus is moved to prev/next item after removal

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### CODE REVIEW

File: packages/@react-aria/tag/src/useTag.ts [hunk 1]
```diff
@@ -56,14 +54,14 @@ export function useTag(props: TagProps<any>, state: GridState<any, any>): TagAri
     focusMode: 'cell'
   }, state, tagRef);
 
-  function onKeyDown(e: KeyboardEvent<Element>) {
+  function onKeyDown(e: KeyboardEvent) {
     if (e.key === 'Delete' || e.key === 'Backspace' || e.key === ' ') {
-      onRemove(children, e);
+      onRemove(gridCellProps['data-key']);
```
Comment (devongovett) [inline packages/@react-aria/tag/src/useTag.ts:59]: You should use `item.key` here instead. Data attributes will be converted to strings, but keys can also be numbers. Same below.

File: packages/@react-spectrum/tag/src/Tag.tsx [hunk 2]
```diff
@@ -59,10 +57,9 @@ export function Tag<T>(props: SpectrumTagProps<T>) {
           styles,
           'spectrum-Tags-item',
           {
-            'is-disabled': isDisabled,
             'focus-ring': isFocusVisible,
             'is-focused': isFocused,
-            'not-removable': !isRemovable,
+            'not-removable': !allowsRemoving,
```
Comment (reidbarber) [inline packages/@react-spectrum/tag/src/Tag.tsx:62]: Don't think this class gets used anywhere, so we can actually remove this line

File: packages/@react-spectrum/tag/test/TagGroup.test.js [hunk 3]
```diff
@@ -208,35 +158,10 @@ describe('TagGroup', function () {
...
-    let inputs = getAllByRole('textbox');
-    expect(document.activeElement).toBe(inputs[0]);
-    userEvent.tab();
-    expect(document.activeElement).toBe(inputs[1]);
-    userEvent.tab({shift: true});
-    expect(document.activeElement).toBe(inputs[0]);
-  });
-
-  it('not disabled if extraneous keys are provided', function () {
+  it('not disabled if keys are provided', function () {
```
Comment (reidbarber) [inline packages/@react-spectrum/tag/test/TagGroup.test.js:161]: IMO we can remove this test since disableKeys won't be in the types

Comment (tiffany-sy-lin) [reply packages/@react-spectrum/tag/test/TagGroup.test.js:161]: I left it as a test to confirm disabled tests don't work anymore, but can remove if you don't think that's necessary

**Review by reidbarber — APPROVED** [review]
LGTM

**Comment by dannify** [comment]
I found 2 issues with focus (which I think we already found?) but they are not blockers for beta or this PR. Please add them to an upcoming sprint.
1. Delete a Tag. Cannot use left/right arrows to navigate right after.
2. Deleting the last tag loses focus (should go to the previous tag)

**Review by dannify — APPROVED** [review]

**Review by devongovett — APPROVED** [review]

**Pull request merged**